tests/backend/checkmach/dune

```diff
@@ -1,18 +1,113 @@
 (rule
  (alias   runtest)
  (enabled_if (= %{context_name} "main"))
- (targets t.o t.cmx s.o s.cmx output.corrected t.cmx.output)
  (deps s.ml t.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
+ (alias   runtest)
+ (enabled_if (= %{context_name} "main"))
+ (deps t5.ml test_assume.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
+ (alias   runtest)
+ (enabled_if (and (= %{context_name} "main")
+                   %{ocaml-config:flambda}
+                  ;; what we really want to say but dune doesn't know about flambda2:
+                  ;; (or %{ocaml-config:flambda} %{ocaml-config:flambda2})
+                  ))
+ (deps test_flambda.ml)
+ (action (run %{bin:ocamlopt.opt} %{deps} -c -alloc-check -O3)))
+
+(rule
```

Comment:
(The duplications in this file are rather
annoying. I wonder whether we should
leave a CR to remember to refactor it
when dune gives us the abstractions
we need.)

Reply:
We can probably use cinaps. I'll add a CR in a different  (followup) PR, which needs to be rebased anyway, to save a round of CI.